client/python/openlineage/client/client.py

```diff
@@ -27,55 +25,48 @@ class OpenLineageClientOptions:
 
 class OpenLineageClient:
     def __init__(
-            self,
-            url: str,
-            options: OpenLineageClientOptions = OpenLineageClientOptions(),
-            session: Session = None
+        self,
+        url: Optional[str] = None,
+        options: Optional[OpenLineageClientOptions] = None,
+        session: Optional[Session] = None,
+        transport: Optional[Transport] = None,
     ):
-        url = url.strip()
-        try:
-            parsed = parse_url(url)
-            if not (parsed.scheme and parsed.netloc):
-                raise ValueError(f"Need valid url for OpenLineageClient, passed {url}")
-        except Exception as e:
-            raise ValueError(f"Need valid url for OpenLineageClient, passed {url}. Exception: {e}")
-        self.url = url
-        self.options = options
-        self.session = session if session else Session()
-        self.session.headers['Content-Type'] = 'application/json'
+        if url:
+            # Backwards compatibility: if URL is set, use old path to initialize
+            # HTTP transport.
+            if not options:
+                options = OpenLineageClientOptions()
+            if not session:
+                session = Session()
+            self._initialize_url(url, options, session)
+        elif transport:
+            self.transport = transport
+        else:
+            self.transport = get_default_factory().create()
 
-        if self.options.api_key:
-            self._add_auth(options.api_key)
-        if self.options.adapter:
-            self.session.mount(self.url, options.adapter)
+    def _initialize_url(
+        self,
+        url: str,
+        options: OpenLineageClientOptions,
+        session: Session
+    ):
+        self.transport = HttpTransport(HttpConfig(
+            url=url,
+            timeout=options.timeout,
+            verify=options.verify,
+            api_key=options.api_key,
+            session=session,
+            adapter=options.adapter
+        ))
 
     def emit(self, event: RunEvent):
-        data = Serde.to_json(event)
-        if log.isEnabledFor(logging.DEBUG):
-            log.debug(f"Sending openlineage event {event}")
-        resp = self.session.post(
-            urljoin(self.url, 'api/v1/lineage'),
-            data,
-            timeout=self.options.timeout,
-            verify=self.options.verify
-        )
-        resp.raise_for_status()
-        return resp
-
-    def _add_auth(self, api_key: str):
-        self.session.headers.update({
-            "Authorization": f"Bearer {api_key}"
-        })
+        if not isinstance(event, RunEvent):
+            raise ValueError("`emit` only accepts RunEvent class")
+        if not self.transport:
+            log.error("Tried to emit OpenLineage event, but transport is not configured.")
```

Comment:
I think we should default the transport to `console`, rather than throw an error and document the behavior. In the `openlineage-java` lib, we default to [`ConsoleTransport `](https://github.com/OpenLineage/OpenLineage/blob/feature/http-client-java/client/java/src/main/java/io/openlineage/client/OpenLineageClient.java#L52)

Reply:
Added `ConsoleTransport` and made it default if there's no valid config or env variable.